modules/apps/notification/notification-web/src/main/java/com/liferay/notification/web/internal/portlet/action/NotificationTemplateObjectFieldMVCResourceCommand.java

```diff
@@ -0,0 +1,124 @@
+import com.liferay.object.model.ObjectField;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectFieldLocalService;
+import com.liferay.portal.kernel.feature.flag.FeatureFlagManagerUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCResourceCommand;
+import com.liferay.portal.kernel.util.LinkedHashMapBuilder;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * @author Paulo Albuquerque
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + NotificationPortletKeys.NOTIFICATION_TEMPLATES,
+		"mvc.command.name=/notification_templates/notification_template_object_field"
+	},
+	service = MVCResourceCommand.class
+)
+public class NotificationTemplateObjectFieldMVCResourceCommand
+	extends BaseNotificationTemplateTermsMVCResourceCommand {
+
+	@Override
+	protected void doServeResource(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		_objectDefinition = _objectDefinitionLocalService.fetchObjectDefinition(
+			ParamUtil.getLong(resourceRequest, "objectDefinitionId"));
+
+		if (_objectDefinition == null) {
+			return;
+		}
+
+		super.doServeResource(resourceRequest, resourceResponse);
+	}
+
+	@Override
+	protected Set<Map.Entry<String, String>> getEntrySet() {
+		List<ObjectField> objectFields =
+			_objectFieldLocalService.getObjectFields(
+				_objectDefinition.getObjectDefinitionId());
+
+		Map<String, String> termValues = new LinkedHashMap<>();
+
+		for (ObjectField objectField : objectFields) {
+			if (StringUtil.equals(objectField.getName(), "creator") &&
+				FeatureFlagManagerUtil.isEnabled("LPS-171625")) {
+
+				termValues.putAll(_creatorTermValues);
+			}
+			else {
+				termValues.put(
+					objectField.getLabel(LocaleUtil.getDefault()),
+					objectField.getName());
+			}
+		}
+
+		return termValues.entrySet();
+	}
+
+	@Override
+	protected String getTermName(String value) {
+		return ObjectDefinitionNotificationTermUtil.getObjectFieldTermName(
+			_objectDefinition.getShortName(), value);
+	}
+
+	private static final Map<String, String> _creatorTermValues =
+		Collections.unmodifiableMap(
+			LinkedHashMapBuilder.put(
+				"Author Email", "creator_email"
+			).put(
+				"Author First Name", "creator_firstname"
+			).put(
+				"Author ID", "creator_id"
+			).put(
+				"Author Last Name", "creator_lastname"
+			).put(
+				"Author Middle Name", "creator_middlename"
+			).put(
+				"Author Prefix", "creator_prefix"
+			).put(
+				"Author Suffix", "creator_suffix"
+			).build());
```

Comment:
Replace to [enum](https://github.com/liferay/liferay-portal/blob/cc505f6d77cc8a53ab9fd21a593ff69d59a8303a/modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/util/ObjectCodeEditorUtil.java#L127) and localize the key.